lib/evm/gas.ex

```diff
@@ -390,6 +390,26 @@ defmodule EVM.Gas do
     @g_call + call_value_cost(value) + gas_limit
   end
 
+  def operation_cost(:log0, [_offset, size | _], _machine_state, _exec_end) do
+    @g_log + @g_logdata * size
+  end
+
+  def operation_cost(:log1, [_offset, size | _], _machine_state, _exec_end) do
+    @g_log + @g_logdata * size + @g_logtopic
+  end
+
+  def operation_cost(:log2, [_offset, size | _], _machine_state, _exec_end) do
+    @g_log + @g_logdata * size + @g_logtopic * 2
+  end
+
+  def operation_cost(:log3, [_offset, size | _], _machine_state, _exec_end) do
+    @g_log + @g_logdata * size + @g_logtopic * 3
+  end
+
+  def operation_cost(:log4, [_offset, size | _], _machine_state, _exec_end) do
```

Comment:
I feel like we could refactor these even more 🤔something like?:

    def operation_cost(:log4, [_offset, size | _], _machine_state, _exec_end) do
        log_cost(4)
    end

    defp log_cost(count)
        @g_log + @g_logdata * size + @g_logtopic * count
    end

Reply:
I thought about it. But in Yellow Paper in cost function there are 4 clauses for each of log operations and I think it's more readable.